### PR TITLE
Allocations module

### PR DESCRIPTION
**Pull request by dcoric**
This PR adds:
- Allocations module from the YuniKorn. 
- Removed test & hello world modules and components
- Updated readme file

### CODE REVIEW

File: web/src/app/allocations-drawer/allocations-drawer.component.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  allocColumnDef: ColumnDef[] = [];
+  allocColumnIds: string[] = [];
+  selectedAllocationsRow: number = -1;
+
+  ngOnChanges(): void {
+    if (this.allocDataSource) {
+      this.allocDataSource.paginator = this.allocPaginator;
+      this.allocDataSource.sort = this.allocSort;
+    }
+    console.log("!", this.externalLogsBaseUrl);
```
Comment (dave-gantenbein) [inline web/src/app/allocations-drawer/allocations-drawer.component.ts:38]: Let's not leave debug code in here.

File: web/src/app/allocations-drawer/allocations-drawer.component.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const result = cpuAndMemoryElements.concat(otherElements);
+
+    return result;
+  }
+
+  isAllocDataSourceEmpty() {
+    return this.allocDataSource?.data && this.allocDataSource.data.length === 0;
+  }
+
+  allocationsDetailToggle(row: number) {
```
Comment (dave-gantenbein) [inline web/src/app/allocations-drawer/allocations-drawer.component.ts:86]: Can we get a unit test for all this nested logic?

**Review by dave-gantenbein — CHANGES_REQUESTED** [review]

File: web/src/app/allocations-drawer/allocations-drawer.component.ts (hunk 1 above)
Comment (dcoric) [reply web/src/app/allocations-drawer/allocations-drawer.component.ts]: Fixed

File: web/src/app/allocations-drawer/allocations-drawer.component.ts (hunk 2 above)
Comment (dcoric) [reply web/src/app/allocations-drawer/allocations-drawer.component.ts:85]: Missing tests are added to the spec file.

**Review by dave-gantenbein — APPROVED** [review]

**Pull request merged**